src/devices/Qmc5883l/Qmc5883l.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Device.Model;
+using System.Numerics;
+using UnitsNet;
+
+namespace Iot.Device.Qmc5883l
+{
+    /// <summary>
+    /// 3-Axis Digital Compass QMC5883L
+    /// </summary>
+    [Interface("3-Axis Digital Compass QMC5883L")]
+    public class Qmc5883l : IDisposable
+    {
+        /// <summary>
+        /// QMC5883L Default I2C Address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x0D;
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// QMC5883L Direction Vector
+        /// </summary>
+        [Telemetry]
+        public Vector3 Direction => ReadDirectionVector();
+
+        /// <summary>
+        /// QMC5883L Heading (DEG)
+        /// </summary>
+        public Angle Heading => VectorExtentsion.GetHeading(ReadDirectionVector());
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Qmc5883l"/> class.
+        /// </summary>
+        public Qmc5883l(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+        }
+
+        /// <summary>
+        /// Sets the sensors mode with additional parameters.
+        /// </summary>
+        /// <param name="mode">Sensor mode</param>
+        /// <param name="interrupt">Controlls the Interrupt PIN</param>
+        /// <param name="rollPointer">When the point roll-over function is enabled, the I2C data pointer automatically rolls between 00H ~ 06H</param>
+        /// <param name="outputRate">Typical Data Output Rate (Hz)</param>
+        /// <param name="oversampling">Bandwidth of an internal digital filter</param>
+        /// <param name="fieldRange">The field range goes hand in hand with the sensitivity of the magnetic sensor</param>
+        public void SetMode(Mode mode = Mode.Continuous,
```

Comment:
So as all those modes are available, why not propose a property for each one?
This will be super easy to adjust only one mode for the developer using this biding. With the specific decoration, that will allow as well to adjust the mode easily. You still can keep this function public. You'll need to call it anyway to set up any change in the modes.